service-registry/registry-zero-config/src/main/java/org/apache/servicecomb/zeroconfig/server/ServerUtil.java

```diff
@@ -179,12 +179,12 @@ private static ServerMicroserviceInstance buildServerMicroserviceInstanceFromMap
 
   private static void startListenerForRegisterUnregisterEvent() {
     try {
-      byte[] buffer = new byte[DATA_PACKET_BUFFER_SIZE];
       multicastSocket = new MulticastSocket(PORT);
       group = InetAddress.getByName(GROUP);
       multicastSocket.joinGroup(group); // need to join the group to be able to receive the data
 
       while (true) {
+        byte[] buffer = new byte[DATA_PACKET_BUFFER_SIZE];
         DatagramPacket receivePacketBuffer = new DatagramPacket(buffer, buffer.length);
         multicastSocket.receive(receivePacketBuffer);
         String receivedPacketString = new String(receivePacketBuffer.getData(), ENCODE);
```

Comment:
This logic seams not correct. Maybe you'd check the length of data actually received and using the correct data.

Reply:
good point. Unfortunately, based on the Multicast working mechanism, we have to create a buffer before data is received.  After data is received, then buffer is filled with the received data.   So we can't create a buffer with size exactly same as the received data.

By the way, it's also mentioned in the JDK source code comments: 
https://github.com/frohoff/jdk8u-jdk/blob/master/src/share/classes/java/net/DatagramSocket.java#L698-L700

Reply:
Maybe you misunderstanding my point. You should do this 

```
 multicastSocket.receive(receivePacketBuffer);
int len = multicastSocket.getLength() ;
if(len > 0) {
   String receivedPacketString = new String(receivePacketBuffer.getData(), 0, len, ENCODE);
}
```